### ISSUE TITLE
Map[String,T] creates imprecise swagger documentation

### ISSUE THREAD

**Issue by hejfelix** [softwaremill/tapir#142]
The swagger generation creates imprecise types for e.g. `Map[String,MyCaseClass]`. The generated swagger.yml looks like so:

```yaml
Map:
  type: object
``` 

for the schema. 

I would expect it to use `additionalProperties`:

```yaml
Map:
  type: object
  additionalProperties:
    $ref: '#/components/schemas/MyCaseClass'
```

**Comment by hejfelix** [comment]
What would this require? Maybe I can find time to do it

**Comment by adamw** [comment]
@hejfelix My guess would be extending `Schema` with a product variant corresponding to an "open" object (but not any open object - one with a fixed field value type). Maybe `Schema.OpenProduct`?

And if an "open" object is detected, generating the appropriate docs.

**Comment by hejfelix** [comment]
So it needs a change in core?

**Comment by adamw** [comment]
Yes, I think that's the cleanest way to do it.

**Comment by adamw** [comment]
Released in 0.9.3

**Comment by hejfelix** [comment]
You are awesome, thanks so much :)
